GTG/core/base_store.py

```diff
@@ -25,17 +25,52 @@
 import logging
 
 from lxml.etree import _Element
-from typing import Dict, List, Optional, TypeVar, Generic, Protocol
+from typing import Dict, List, Optional, TypeVar, Generic
+from typing_extensions import Self
 
 
 log = logging.getLogger(__name__)
 
-S = TypeVar('S',bound='Storeable')
+S = TypeVar('S',bound='StoreItem')
+
+class StoreItem(GObject.Object):
+    """Base class for items in BaseStore."""
+
+    __gtype_name__ = 'gtg_StoreItem'
+
+
+    def __init__(self,id: UUID):
+        self.id: UUID = id
+        self.parent: Optional[Self] = None
+        self.children: List[Self] = []
+        super(StoreItem, self).__init__()
+
+
+    @GObject.Property(type=int)
+    def children_count(self) -> int:
+        """Read only property."""
+        return len(self.children)
+
+
+    @GObject.Property(type=bool, default=False)
+    def has_children(self) -> bool:
+        return len(self.children) > 0
+
+
+    def get_ancestors(self) -> List[Self]:
+        """Return all ancestors of this tag"""
+        ancestors: List[Self] = []
+        here = self
+        while here.parent:
+            here = here.parent
+            ancestors.append(here)
+        return ancestors
+
+
+    def check_possible_parent(self, target) -> bool:
```

Comment:
```suggestion
    def is_parent_of(self, target) -> bool:
```
and inverse the conditions below.

I think it might be clearer to read `if not foo.is_parent_of(bar):` rather than `if foo.check_possible_parent(bar):`?

Reply:
I agree that the name is not perfect. However, `is_parent_of` wouldn't correctly describe the method (e.g., you can not parent an item to itself.) I used `is_parentable_to` in the new version. I think it also reads much better than the original one.